File: ws_perf.old/producers/pub.js
var zmq = require('zmq')
  , sock = zmq.socket('pub');

var i = 0,
    fps = 100;

sock.bindSync('tcp://127.0.0.1:5556');
console.log('Publisher bound to port 5556');

setInterval(function(){
  console.log('sending a multipart message envelope');
  var mess = {"t": new Date(), "y": Math.cos(i/100.0)};
  var stringObject=JSON.stringify(mess);
  i += 1;
  sock.send(['data', stringObject]);
}, 1000.0/fps);

File: ws_perf.old/producers/pub.py
import time
import math
import json
import zmq

c = zmq.Context()
s = c.socket(zmq.PUB)
s.bind("tcp://127.0.0.1:5556")
print('Publisher bound to port 5556')

fps = 100.0
i = 0.0
while 1:
    mess = {"t": time.time()*1000, "y": math.cos(i/100)}
    print(mess)
    s.send_multipart(["data ", json.dumps(mess)])
    time.sleep(1.0/fps)
    i+=1